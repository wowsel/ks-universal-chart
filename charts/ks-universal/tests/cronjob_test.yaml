suite: test cronjob
templates:
  - cronjob.yaml
tests:
  - it: should create a cronjob
    set:
      cronJobs:
        test-backup:
          schedule: "0 2 * * *"
          containers:
            backup:
              image: backup-tool
              imageTag: "1.0.0"
              command: 
                - /bin/sh
                - -c
                - backup.sh
              env:
                - name: BACKUP_DIR
                  value: /backup
    asserts:
      - isKind:
          of: CronJob
      - isAPIVersion:
          of: batch/v1
      - equal:
          path: metadata.name
          value: test-backup
      - equal:
          path: spec.schedule
          value: "0 2 * * *"
      - matchSnapshot:
          path: spec.jobTemplate.spec.template.spec 