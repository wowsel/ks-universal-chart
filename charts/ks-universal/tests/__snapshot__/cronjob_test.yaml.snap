should create a cron job with specified configuration:
  1: |
    containers:
      - command:
          - /bin/sh
          - -c
          - backup.sh
        env:
          - name: BACKUP_DIR
            value: /backup
        image: backup-tool:1.0.0
        name: backup
    restartPolicy: Never
